text/0000-monorepo.md

```diff
@@ -0,0 +1,359 @@
+
+```
+# Team A
+carbon-addons-a
+carbon-addons-b
+carbon-addons-c
+```
+
+In the existing landscape, team B would need to install (at minimum) the following:
+
+```bash
+yarn add carbon-components \
+  carbon-components-react \ 
+  carbon-icons \
+  carbon-addons-a \
+  carbon-addons-a-react \
+  carbon-addons-b \
+  carbon-addons-b-react 
+```
+
+Ideally, the best experience for this group would be a single `carbon-addons-b` or `carbon-addons-b-react`.
+
+The other downside to this is the aspect of who owns support and who decides what is in-scope for enhancements of a component. When components are built inside add-ons, this can fall on the team whose add-on houses the component. However, if another team is requesting enhancements most likely they should be the ones to implement (when reasonable).
+
+As a result, the proposal is that **all** component development happens in the Carbon repo and add-on packages move to becoming **the defacto way to use Carbon for a Business unit, portfolio, offering, or team**. What this means is that add-ons move increasingly towards their domain, and components like `PlatformHeader` should move to the main Carbon repo.
+
+This does not prohibit multiple `PlatformHeader` components, however. If we used the situation above, we could have a singular `platform-header` package. Inside this package we could have multiple implementations under `PlatformHeader.A.js`, `PlatformHeader.B.js`, etc. This allows us to both allow teams to ship things under their unique deadlines/constraints while also allowing colocation of related components with the eventual goal of converging implementations.
+
+In addition, having shared packages allows teams to make proposals through a formal RFC process and decide on changes as a group instead of tying ownership to any particular add-on.
+
+As a result, the add-on packages increasingly move towards providing Carbon in the best way for their group. This can include depending on the entire component library, or only on the components a brand needs. In addition, add-ons would provide domain-specific behavior (like data fetching) alongside components to make it even easier for product developers to deliver new functionality.
+
+## GitHub
+
+The proposed project will live in GitHub and subsequently we will need to consider how this impacts things like:
+
+- Notifications
+- Issue templates
+- Pull Request templates
+
+These areas will need to be addressed in order to prevent excessive notifications, and provide reasonable options for issue templates to help with triage. Pull request templates are one item that should be shared across the projects as it will provide a formal structure for documenting proposed changes and reviewing them.
+
+For Notifications, most likely we would have to do a custom solution instead of relying on GitHub notifications. There are a couple strategies we could follow:
```

Comment:
Definitly need to look at hooking up a bot to slack ... I'd love to get slack notifications for issues I'm mentioned in or things I'm codeowner for.

We definitely need some small cross cutting channels as well ...